### PR TITLE
Don't allow people to add their own usernames to the codebase as such

### PR DESCRIPTION
**Pull request by mrhwick**
Changelog:
- Added a function `owner_name_added_in_diff()` which will detect the PR owner's name in the PR diff as an addition.
- Reject PRs outright that exhibit selfishness.

### CODE REVIEW

File: github_api/prs.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import arrow
+import re
+import requests
```
Comment (reddraggone9) [inline github_api/prs.py]: This import doesn't appear to be used anywhere.

File: cron/poll_pull_requests.py [hunk 2]
```diff
@@ -23,6 +23,16 @@ def poll_pull_requests():
         pr_num = pr["number"]
         __log.info("processing PR #%d", pr_num)
 
+        # If the owner of this PR added their name to the codebase in any way.
+        if gh.prs.owner_name_added_in_diff(api, settings.URN, pr_num, pr['user']['login']):
+            # We will reject the PR with a message about selfishness not being allowed.
+            __log.info("PR %d rejected for selfishness, closing", pr_num)
+            gh.comments.leave_reject_comment_too_selfish(api, settings.URN, pr_num)
+            gh.prs.label_pr(api, settings.URN, pr_num, ["rejected"])
+            gh.prs.close_pr(api, settings.URN, pr)
```
Comment (reddraggone9) [inline cron/poll_pull_requests.py:32]: You could factor out a reject_with_comment function instead of duplicating this logic.

**Review by reddraggone9 — CHANGES_REQUESTED** [review]

File: github_api/prs.py (hunk 1 above)
Comment (mrhwick) [reply github_api/prs.py]: I'll remove this

File: cron/poll_pull_requests.py (hunk 2 above)
Comment (mrhwick) [reply cron/poll_pull_requests.py:32]: I'll factor this out in a future refactoring PR

**Comment by PlasmaPower** [comment]
:-1: doesn't stop anything malicious while still preventing PRs like #166

**Pull request closed without merging**